### ISSUE TITLE
Ethereum Core Devs Meeting 23 Agenda

### ISSUE THREAD

**Issue by Souptacular** [ethereum/pm#21]
# Ethereum Core Devs Meeting 23 Agenda
### [YouTube Live Stream Link](https://www.youtube.com/watch?v=PQjeAZyL2_w)
# Agenda

**Reminder: Metropolis is now split into 2 hard forks: "Byzantium" first and then "Constantinople".**

1. Metropolis updates/EIPs.
  **a. Any "subtleties" or questions we need to work out.**
      - JSON RPC updates needed by the receipt status change EIP. [Peter]
      - [EIP 649: Metropolis Difficulty Bomb Delay and Issuance Reduction](https://github.com/ethereum/EIPs/pull/669): Community concerns need to be addressed [Hudson]
  **b. Updates to testing.**
  **c. Details and implementations of EIPs.**
        1. Updates from client teams.
            - geth - https://github.com/ethereum/go-ethereum/issues?q=label%3Ametropolis+is%3Aclosed
            - Parity - https://github.com/paritytech/parity/issues/4833
            - cpp-ethereum - https://github.com/ethereum/cpp-ethereum/issues/4050
            - ethereumJ
            - yellowpaper -  https://github.com/ethereum/yellowpaper/issues/229
            - pyethapp
            - Other clients
        2. Determining gas prices for new opcodes & pre-compiles [Martin HS/Arkadiy]
  **d. Review time estimate for testing/release.**

Please provide comments to add or correct agenda topics.

**Comment by TheCryptoMines** [comment]
Hi Souptacular,

Can we have the item of Issuance reduction added to the Agenda please? I (and many many others) would like a definitive reasoning for the reduction to be presented by the Dev team. We have seen numerous community members discuss why this should / should not happen but the only actual reasoning from the Dev team was EIP 186. The reasoning in this EIP was to use this as a tool to increase the Ethereum price to help both Investors and the Dev team for funding. The price of ETH when this EIP was created was $7.82, right now it is trading at $289.55, or and increase of 3602%.

The latest updates for EIP 669 indicate the new motivation for this change is "The Casper development and switch to Proof-of-Stake is delayed, the Proof-of-Work should be feasible for miners and create new blocks every 15 seconds on average for another 1.4 years. The incentive to continue mining the Proof-of-Work chain after a potential move to Proof-of-Stake is reduced by adjusting block rewards." 

This makes no sense - there is already a mechanism in place to incentivise Miners to move to POS and it is the Difficulty Bomb, it has been delayed due to Casper delay but should come back into play as we get closer to the POS release. The statement in EIP 669 for the motivation seems to be just put there as the Dev team have no real reasoning to reduce the Issuance?

I understand that the Carbonvote took place but it heavily favoured large ETH holders who have a huge amount of self interest in reducing the mining rewards to try and inflate the price. Therefore, I do not believe the results of this vote can be used as an objective way in showing the community preference.

Thank you for taking the time to review this and I look forward to hearing from the team their reasoning behind this EIP continuing.

Stephen

**Comment by DeviateFish-2** [comment]
Posted some comments on the EIP: https://github.com/ethereum/EIPs/pull/669#issuecomment-323178761

Would be nice to get some explicit rationale around why issuance is being reduced, without falling back to any of the (already disproven) lines of reasoning...  Though to be honest, even a statement of what problem this is solving, and why, would be nice.  So far, it's been "we're reducing issuance", with no mention of why that's even a desirable change (much less happening at all).

**Comment by DeviateFish-2** [comment]
I see it's been approved ([as of 7 days ago](https://github.com/ethereum/EIPs/pull/688)), regardless of any discussion.

So, why again have people been asked to submit feedback, if it's been confirmed that an issuance reduction is happening, without any reasoning or explanation?  There's not even an explanation for why the change is needed, much less wanted, nor any explanation for the magnitude of the reduction.

Is this to be expected for all future protocol changes?

**Comment by Souptacular** [comment]
@DeviateFish2
It was approved in one of the previous core dev meetings that were recorded. It is being brought up in the meeting this Friday for clarification from those, including you, in the community who have shown concern about why it has been an agreed upon change.

**Comment by DeviateFish** [comment]
Yeah, see, that's my concern.

Clarification != discussion, nor does it imply the possibility that said change will be reverted.

Given that the devs involved in the discussion so far have done everything in their power to avoid actually answering the questions or giving any rationale (instead preferring to, put simply, troll), it's become more and more apparent that there's no intention of changing course, regardless of any consensus or lack thereof.

Which, again, begs the question: why ask for input and pretend there will be any effect on the direction of the protocol if it's just going to be a huge waste of time for everyone involved?

**Comment by DeviateFish-2** [comment]
(wrong account)...

@Souptacular Yeah, see, that's my concern.

Clarification != discussion, nor does it imply the possibility that said change will be reverted.

Given that the devs involved in the discussion so far have done everything in their power to avoid actually answering the questions or giving any rationale (instead preferring to, put simply, troll), it's become more and more apparent that there's no intention of changing course, regardless of any consensus or lack thereof.

Which, again, begs the question: why ask for input and pretend there will be any effect on the direction of the protocol if it's just going to be a huge waste of time for everyone involved?

[E] Still waiting for this to be officially added to the agenda, above.

**Comment by Souptacular** [comment]
@DeviateFish-2 
It has been added to the agenda. Your concerns are noted. There is an open invitation for you to join the meeting tomorrow, but I believe you previously said you were not able to.

**Comment by DeviateFish-2** [comment]
@Souptacular  Thanks

And that's correct, the time isn't great for me.  I think there might be some others wanting to join... how does one join the call, anyway?

**Comment by Souptacular** [comment]
@DeviateFish-2

If they are heavily involved in developing an Ethereum core client, a piece of the Ethereum protocol (network, testing, etc.), or are involved in an EIP that is going to be discussed on the core devs call they can ask me about joining a call and I will send them the Google hangouts room info.

I want to make sure I am accurately framing your concerns and what you want accomplished.

1. You want a reason why EIP 669 was approved in the first place.
2. You feel that "there is no rational basis for issuance reduction that isn't a rationalization designed to support the issuance reduction itself" and would like there to be a solid reason for the network to perform an issuance reduction.
3. You reject the  "overpaying for security" argument and the "miners being too powerful" arguments as reasons to support an issuance reduction.
4. You feel that ultimately a problem needs to be pointed out that an issuance reduction would solve, and not the other way around.

I am planning on using the above to frame the discussion, but am happy to edit it if I have not correctly listed your concerns.

**Comment by DeviateFish-2** [comment]
@Souptacular Yep, that's pretty accurate, though a little bit of refinement, because I know if it's frame exactly like you put it what the answers will already be.

One refinement would be that I want a reason why EIP 186 was merged into EIP 649 in the first place.  EIP 649 (e.g. delaying the Ice Age) I have 0 issues with, and I don't want to give the opportunity for a cheap deflection onto the Ice Age removal instead of an answer about the issuance reduction itself.

Second, I want to know the rationale behind a reduction from 5 ETH per block to 3 ETH per block, why 3 ETH, and, even more specifically, why everyone involved has specifically avoided answering that question.  Saying "because it's roughly equivalent to the current issuance" is not an answer, it's avoiding an answer entirely, because it refuses to answer why a reduction from 5 ETH to 3 ETH is desired or even acceptable in the first place.

Point 1 is important because without an explanation, it just furthers the evidence that Ethereum is absolutely centralized, to the point where governance decisions no longer even need to be explained before being imposed on the network.

Points 2 and 4 above, though, are easily the most important points.  I firmly believe that those in favor of an issuance reduction haven't thought further than "more restricted supply = higher ETH price" and "more restricted supply = more early adopter staking power", and are ignoring or neglecting the side effects of a lower ETH issuance.

Three that I can name off the top of my head:

1. Lower uncle tolerance.  Given that tolerance for uncles is proportional to `uncle reward / (block reward + fees)`, which can be refactored into `.75 / (1 + fees / block reward)`, lowering the block reward will lower the tolerance for uncles.
2. More miner centralization pressure.  The first people to be impacted by a reduced issuance will be solo miners and small hobby miners.  This will lead to pools like f2pool (who has already demonstrated a predisposition for acting against the network) to have an increased share of the hashpower.  Given that they already control enough of the network to have some small success with selfish mining attacks, centralization pressure on miners will just give them even more power in that regard.
3. More staking centralization.  A lot of the modeling around PoS doesn't take into account the vast discrepancy between how much was paid to acquire the ETH used to stake, but assumes that all ETH are worth the same to everyone.  This change amplifies that discrepancy, by making it harder to dilute the existing pre-sale stake.  It's roughly equivalent to starting a PoW coin wherein 70+% of the miners bought their entire farms for pennies, but all those who enter after them must pay market prices for their hashpower--and then making a change that allows for even fewer of those new miners to even join.  Reducing future issuance just increases the power discrepancy for the future PoS world.

The last one is especially dangerous; the fact that it's being actively promoted and pursued raises some pretty serious concerns about Ethereum's future.

**Comment by vbuterin** [comment]
> Lower uncle tolerance. Given that tolerance for uncles is proportional to uncle reward / (block reward + fees), which can be refactored into .75 / (1 + fees / block reward), lowering the block reward will lower the tolerance for uncles.

This effect is negligible. Currently, fees are ~400 ETH per day, and with the 3 ETH reduction plus difficulty bomb delay block rewards will be ~18000 ETH per day. Even in bitcoin, fees are still only ~20% of miners' reward.

> More miner centralization pressure. The first people to be impacted by a reduced issuance will be solo miners and small hobby miners.

OTOH having a _higher_ miner reward may increase centralization because it increases the incentive to develop an ASIC.

> More staking centralization

Do we have evidence that PoW rewards increase decentralization of coin holdings on net? This could actually go both ways. I'd argue that it has increased decentralization _so far_, but in the future, the surplus may well go to increasingly professionalized mining farms, regardless of what the reward is.

**Comment by vbuterin** [comment]
The reason for the reduction in issuance is IMO just the simple first-order argument. It saves the Ethereum ecosystem ~$4m per day, as well as reducing externalities such as global warming and our effects on GPU prices. This comes at a very low cost to security, because Ethereum already has a near-hegemony on GPU miners.

The reason to reduce to 3 ETH/block specifically has two arguments:

1. The political argument. Given that block times during the fork will be ~25-30 sec, 3 ETH preserves the per-second block reward from before the fork, which means that Byzantium as a whole is neutral toward miners' interests (arguably positive because it prevents the ice age from going further). This is a Schelling point, which is good because it makes it much easier to coordinate around this specific value as opposed to others. Also, while I agree it's only one indicator among many, it's [roughly what the carbonvote wants](http://carbonvote.com/).
2. The technical argument. Because of the ice age, we already have mining rewards approaching ~0.21 ETH/sec (equivalent to 3 ETH per 14s block). Hence, anything equal to the above this value is to some extent a known quantity, anything below this value is an unknown. This means that the risks at these levels are lower than the risks of pushing mining rewards down even further (say, to Vlad's 1.25 ETH).

**Comment by vbuterin** [comment]
Ice age update:

* 30 seconds on Sep 22 (4300000)
* 39 seconds on Oct 27 (4400000)

**Comment by DeviateFish-2** [comment]
> This effect is negligible. Currently, fees are ~400 ETH per day, and with the 3 ETH reduction plus difficulty bomb delay block rewards will be ~18000 ETH per day. Even in bitcoin, fees are still only ~20% of miners' reward.

I don't think that it can be so quickly written off as negligible.  Gas used has increased nearly an order of magnitude over the past few months, yet the median gas price hasn't budged.  During peak traffic times, that's on the order of 0.1 ETH per block, which is not even accounting for the increased fees that come with increased traffic.  Reducing the uncle reward by 40% is going to reduce the tolerance for uncles by a similar amount.  Either gas limits will go down, or gas prices will have to go up to accommodate.

> OTOH having a higher miner reward may increase centralization because it increases the incentive to develop an ASIC.

Which is kind of a non-issue for Ethereum, given that it won't stay PoW long enough for that to even become a return on investment of any kind.  This has always been an invalid argument, and only becomes more invalid the closer it gets to PoS.  In other words, this is a red herring.

> Do we have evidence that PoW rewards increase decentralization of coin holdings on net? This could actually go both ways. I'd argue that it has increased decentralization so far, but in the future, the surplus may well go to increasingly professionalized mining farms, regardless of what the reward is.

The surplus can go into professional mining farms all it wants, that's kind of irrelevant.  Miners can't stop a switch to PoS, so at best, they can attempt to recoup any remaining investment on other coins.  Again, sort of a red herring.  The more liquidity the market has, the harder it is for established holders to maintain a chokehold on it.  Thus, the more coin that is generated and sold, the less the supply is being controlled by established holders.

You're also completely sidestepping the issue of presale holders having an inordinate amount of future staking power for cheap--and also completely ignoring the issue of small total issuance leading to even more gains on their part.

I feel like you're deflecting.

> The reason for the reduction in issuance is IMO just the simple first-order argument. It saves the Ethereum ecosystem ~$4m per day, as well as reducing externalities such as global warming and our effects on GPU prices. Mining is already consuming as much electricity as Cyprus and having a significant effect on gamers' ability to get GPUs. Reducing this by 40% comes at a very low cost to security, because Ethereum already has a near-hegemony on GPU miners.

This is a) incorrect, and b) misleading.  For one, it doesn't "save" the ecosystem anything.  According to everyone's [favorite website](http://www.flippening.watch/) mining rewards account for less than 1% of the total daily volume of ETH traded.  Mining has absolutely no impact on how much the "market is paying for security", and as I've laid out many times in the past, quantifying security in terms of $ is faulty.  Security should only be quantified in terms of ETH--the only place $ should come into it is within the mechanism under which miners compete for that ETH.

Further, those externalities won't be reduced at all.  Miners in the GPU space do not turn off their miners when Ethereum becomes unprofitable--they simply switch to the next-most-profitable thing.  Again, misleading and incorrect.

With regards to your comment about Ethereum's dominance of the GPU mining space: Just pulling some quick numbers from [whattomine](https://whattomine.com/), Ethereum accounts for 66% of the top 20 most profitable coins at the moment.  That percentage gets a little more diluted the more coins you pull into this calculation, but it's a very long-tailed distribution.  Figured it would help to have some numbers, though.

However, again, Ethereum's dominance over the GPU space comes from its extreme increase in price--it's not exactly safe to base security decisions around something a volatile as price.  Should ETH's price take a dive, suddenly the network will be left with very few miners at all.  Granted, attacks on the platform would also become far less profitable; but then, this is why security should only be quantified in terms of the current itself.  If security can be quantified in terms of the coin alone, the measure of security becomes completely decoupled from price, and price only becomes a tool that ensures that as the value of an attack goes up, so too does security.

Attempting to couple the two together puts you in a situation where it is easy to underestimate (or overestimate) the amount of security required.

> The political argument. Given that block times during the fork will be ~25-30 sec, 3 ETH preserves the per-second block reward from before the fork, which means that Byzantium as a whole is neutral toward miners' interests (arguably positive because it prevents the ice age from going further). This is a Schelling point, which is good because it makes it much easier to coordinate around this specific value as opposed to others. Also, while I agree it's only one indicator among many, it's roughly what the carbonvote wants.

You and I both know the carbonvote is the worst possible measure of consensus on this issue.  The more coin a person holds, the more this change benefits them--so of course they'll vote for it.  The carbonvote is completely tainted by selection bias, and *you know this*.  Again, I feel like you're deflecting here.

> The technical argument. Because of the ice age, we already have mining rewards approaching ~0.21 ETH/sec (equivalent to 3 ETH per 14s block). Hence, anything equal to the above this value is to some extent a known quantity, anything below this value is an unknown. This means that the risks at these levels are lower than the risks of pushing mining rewards down even further (say, to Vlad's 1.25 ETH).

Your answer, once again, still doesn't answer the question of *why it should be reduced at all*.  All it answers is the question of whether or not a reduction to 3 ETH could be considered safe from a security standpoint.  Which is, again, debatable.  Just because an attack hasn't happened, doesn't mean it can't.  I know if I had a plan to attack the network, but knew that security would be going down in the near future, I'd hold off on doing it until after the security was reduced, because the ability to respond would be similarly reduced.  Hell, malicious actors that would benefit from reduced security would be among the loudest defending this change.

You're not giving malicious actors enough credit, instead oversimplifying the situation as if there is no sense of future or past.

So, like I keep asking:  What is the argument for reducing issuance from 5 ETH per block to 3 ETH per block?  Not "why is it safe to reduce it?" or "will there be harm in reducing it?", but "why reduce it at all?"

It's the "if it isn't broke, don't fix it" argument--and you keep saying you're fixing it, but you won't say what's broken about it.

**Comment by holiman** [comment]
> but assumes that all ETH are worth the same to everyone.

> the issue of presale holders having an inordinate amount of future staking power for cheap

These comments sound very strange to me. There is a market price for ether. Early ether is worth the same as ether bought yesterday. You can't determine the 'worth' of something based on the cost of aquisition a long time ago, when there's a functioning market now. 

>> The political argument. Given that block times during the fork will be ~25-30 sec, 3 ETH preserves the per-second block reward from before the fork, which means that Byzantium as a whole is neutral toward miners' interests (arguably positive because it prevents the ice age from going further). This is a Schelling point, which is good because it makes it much easier to coordinate around this specific value as opposed to others. Also, while I agree it's only one indicator among many, it's roughly what the carbonvote wants.

> You and I both know the carbonvote is the worst possible measure of consensus on this issue. The more coin a person holds, the more this change benefits them--so of course they'll vote for it. The carbonvote is completely tainted by selection bias, and you know this. Again, I feel like you're deflecting here.

You totally ignored a block of text, and answered only to "Also, while I agree it's only one indicator among many, it's roughly what the carbonvote wants.". 

> but "why reduce it at all?"

Vitalik answered that: 

> It saves the Ethereum ecosystem ~$4m per day, as well as reducing externalities such as global warming and our effects on GPU prices.
> Reducing this by 40% comes at a very low cost to security

**Comment by DeviateFish-2** [comment]
> These comments sound very strange to me. There is a market price for ether. Early ether is worth the same as ether bought yesterday. You can't determine the 'worth' of something based on the cost of aquisition a long time ago, when there's a functioning market now.

Yeah, except that's not actually how it works.

Who's more willing to spend 1 ETH?  The one who bought 1k ETH for $300, or the one who bought 1 ETH for $300?  One is spending 0.1% of their investment, the other is spending 100%.

> You totally ignored a block of text, and answered only to "Also, while I agree it's only one indicator among many, it's roughly what the carbonvote wants.".

Because the rest of it, like other points, were red herrings?

Saying it's "neutral to miner interests" completely disregards the fact that the Ice Age itself (and the issuance reduction that comes as a side effect of it) is definitely not "neutral to miner interests".  Again, by framing it as "oh, we're not changing anything" ignores the fact that they're actually changing it to "lock in" the effects of the Ice Age to date.  Framing it in this way is misleading and dishonest.

Also, calling something a "Schelling point" doesn't make it a Schelling point.

> It saves the Ethereum ecosystem ~$4m per day, as well as reducing externalities such as global warming and our effects on GPU prices.
> Reducing this by 40% comes at a very low cost to security

That's not an answer, because again, it frames it in a misleading way.  It only "costs ~$4m per day" because ETH happens to be worth some amount of $.  If ETH were worth 1/100th of that, it would only be "~$40k per day", which would then seem very cheap for the given transaction volume.  The only *honest* way to quantify security is in ETH, and make the case that x ETH is secured for y ETH.

This also doesn't even get into the invalid assumption that somehow miner expenses = 100% of the ETH they make--which is pretty far from true, even now.  ETH is very profitable.

It also implies that the amount of ETH that miners sell somehow has an impact on the $ value of ETH... which again, is far from true.  The daily amount of ETH mined is less than 1% of the daily amount of ETH traded.  There's non correlation at all.

**Comment by kybarnet** [comment]
@vbuterin @Souptacular  I would like to be added to the Call. I will PM my gmail.

I have no idea why financial supporters have not been afforded any voice in the issuance discussion and only verbal arguments of the contractors. This is pay scale discussion, the ones giving the payments are just as important as the ones getting paid.

I have stated numerous times this discussion needs be had in Dollars (or %), for clarity. Here Vitalik converts the effect in US Dollars

>  It saves the Ethereum ecosystem ~$4m per day, as well as reducing externalities such as global warming.

This is correct, and well stated. However, there is a problem.

On July 14th, Vitalik suggested it to be $600 per Block or 3 ETH @ $200 Value

https://github.com/ethereum/EIPs/pull/669#issuecomment-315380512

Ethereum is now valued at $330, and 3 ETH amounts to $990, or 50% more than what Vitalik initially stated. Why was $600 per block OK July 14th, but now August 24th it should be $990??

And I'm not trying to split hairs, but that's over a 50% increase in the payouts since the discussion began, because the argument was not phrased in terms of Dollars, which is the currency used to purchase mining Equipment. Vitalik clearly states it saves $4 Million per day vs other systems, thus ETH clearly has a Dollar value that Vitalik at least recognizes, so I don't want the issue that 'ETH has no Dollar value' continuously brought up.

3 ETH = $2 Billion dollars. I want this to be clearly stated on the call, that on the Annualized basis, Ethereum will spend $2 Billion dollars on miner bonuses at $330 and $4 Billion at $660 values.  This is not 'reward'. Reward includes transaction fees which this does not.  $2 Billion dollars in bonuses, plus transaction fees, equal block reward.

Bitcoin has 130% more market value but only 40% more Miner Bonus cost. How is Bitcoin able to be so much more Dollar cost efficient than Ethereum?

Vitalik states that "1.25 ETH" is an 'unknown', one Vlad supports, but provides no numerical argument against this. In his previous statement, he suggested $600 per block, which is 1.8 ETH. Bitcoin already uses percentages as low as 4%.

How is it that Bitcoin is able to spend just 4% toward miner bonuses, but Ethereum is considered an 'unknown' below 6.7%? Isn't it fair to say that it is a known that miner bonuses as low as 4% are market tolerable?

How is it that in absolute terms, $600 per block is tolerable on July 14th, but on August 25th, it must be $990?

These are million and billion dollar decisions that the Ethereum foundation is required to make, and I work with a significant portion of those invested in Ethereum. I have worked performing economic analysis for other block chains, hired as an economist. I would like to assist the Ethereum foundation, and would like Vlad to be in on the call.

By agreeing to unlimited values of Rent, instead of Dollars of Rent, Ethereum is being unclear about its economic intents. It has suggested numerous times it would like to remove miner bonuses and 51% hash attack from the equation all together.

> So why is it that Bitcoin can pay 4%, but Ethereum refers to it as an 'unknown' to pay less than 6.7%? It's clearly a known, market accepted value to pay 4%.

> Likewise, why is it that $600 is OK on July 14th, but $990 is the amount on August 25th? Why did the estimated cost of Rent go up over 50%?

First $600 is suggested, and now it is presented as if $990 is the agreed upon solution.

First Bitcoin does 4%, and now it is presented as if less than 6.7% is an unknown.

At similar market values, Ethereum would pay $2 Billion more in annual Rent than Bitcoin pays for the same service. Ethereum is currently paying over a Billion dollars more than anyone else, excluding Bitcoin. Even with Bitcoin included, Ethereum only pays marginally less. In absolute terms, Ethereum is higher than virtually everyone, and as a percentage it is significantly higher than Bitcoin.

If $4 Million per day is the 'savings' versus doing nothing, what could be achieved by doing something mathematically reasonable? $6 Million per day? An additional 'savings' of $60 Million per month?

I believe you and your listeners will find the time well spent. I hope to make the call.

kybarnet

**Comment by karalabe** [comment]
@Souptacular Please add a note from my part for discussing JSON RPC updates needed by the receipt status change EIP.

**Comment by karalabe** [comment]
@Souptacular Please add another subtlety note around https://github.com/ethereum/EIPs/issues/684. The EIP only states that "creation exits with an exception" from CREATE if address collision is reached, but it's not clear how much gas is used.

 * The consensus tests currently seem to consume all gas from the outer scope, even if only 63/64th gas is allowed for the CREATE opcode starting from EIP 150.
 * Depth limit and insufficient account balance errors consume only the gas allowance of CREATE (i.e. only 63/64th starting from EIP 150), allowing the calling context to react on the error.

I'd suggest unifying the two behaviors and only consuming 63/64 gas for collisions too.

**Comment by vbuterin** [comment]
I agree with unifying the behaviors of insufficient sender balance and nonempty destination code/nonce for CREATEs.

**Comment by TheCryptoMines** [comment]
Hi @vbuterin 

I believe @DeviateFish-2 has covered many areas we feel strongly about with regards the Issuance Reduction but I'd like to comment on the reasons you have given for the Issuance Reduction both here and in the Reddit thread during the week:

> Given that block times during the fork will be ~25-30 sec, 3 ETH preserves the per-second block reward from before the fork, which means that Byzantium as a whole is neutral toward miners' interests (arguably positive because it prevents the ice age from going further). This is a Schelling point, which is good because it makes it much easier to coordinate around this specific value as opposed to others.

This and many of the other arguments put forward revolve around Miner rewards being the same after the update as they are when the update takes place due to the current slower block rewards caused by the Ice Age.

This is an extremely underhanded way of trying to add rationale to this change. The Ice Age was introduced into the network with one purpose, to help transition miners away from the POW chain after POS was introduced. When it was implemented, the expected timeframe for that to happen was mid to late 2017.

As soon as POS was confirmed to be delayed, the Difficulty Bomb should have been delayed also as it served no purpose. Had it not been for the negligence of the Dev Team in doing this, we would not be having this conversation. It was more than apparent back in May of this year that a full POS implementation would not go ahead in 2017 and had you delayed the Ice Age then, block times would still be at 15s. This delay is also starting to see other impacts on the network such as full blocks and slower transactions times as transaction volume continues to increase. This will again lead to the need for a Gas Limit increase which miners will take the hit on to do for the good of the network.

Trying to use the current block times, caused by an artificial increase in difficulty for a completely different purpose, to justify reducing the rewards is outrageous. A cynic may even suggest that this was done purposefully, specifically for this reason.

> Also, while I agree it's only one indicator among many, it's roughly what the carbonvote wants.

Yes, one indicator heavily biased towards the part of the community who gain the most from this reduction going ahead. I myself ran a poll in the Mining community over the last week of which the results can be seen here: http://www.easypolls.net/poll.html?p=5995b58be4b0110f9f780304. If you are going to reference the CarbonVote for providing reasoning for the change, then I would like this poll to be referenced also.

In addition, during your reddit statements the week, you said you would only support an Issuance Reduction if it had the consensus of the Community. This current proposal does not have community consensus.

Lastly, I asked the mining community to provide some reasoning for leaving the issuance as is, I would like you to cover these also:

- Ethereum's inflation rate is not excessively high for a cryptocurrency that is only a few years old. Bitcoin's was much higher at the equivalent stage in its life.
- Ethereum's inflation structure is already disinflationary.
- Useful currencies tend to be inflationary because inflation encourages people to use rather than hoard currency. Does the Ethereum Foundation want Ether to be used or hoarded?
- Miners and speculators often have different incentives, but miners are the ones who, in my opinion, contribute far more to the network. I think that the policies of the Foundation should consider their interests over those of speculators when there are disputes.
- The purpose of block rewards is not just security. Block rewards help with currency distribution, liquidity, and price discovery. Only about 23% of Ethereum was mined, with the rest having been distributed in the ICO. Keeping the block reward at 5 for the near future would help ensure a fair distribution of Ether for new investors who are interested in the technology but who don't want to buy from speculators who wish to drive up the price.

@Souptacular While I don't believe I fall into the criteria for attending the call, I am available to attend if needed and will be watching the live stream.

**Comment by pirapira** [comment]
@karalabe @vbuterin the gas consumption issue sounds like a bug in `cpp-ethereum`.

**Comment by M41** [comment]
There are arguments on both sides, that 3 ETH per block is too little, or 3 ETH per block is too much. Not surprisingly they are in line with financial interests of miners and hodlers/investors, everyone arguing based on what outcome they want to achieve.  **I am calling what they call rational arguments as bogus**, ultimately they are after financial gains.  

Dev team working hard on all this needs to make a final decision and draw a line in the sand and move forward and away from petty arguments.  No justification to stall or waste any more time on this with the long road ahead and the ambitious timelines and dev projects awaiting.

**Comment by karalabe** [comment]
@Souptacular You might want to link this for Geth's metro progress: https://github.com/ethereum/go-ethereum/issues?q=label%3Ametropolis+is%3Aclosed instead of that stale PR from Jeff.

**Comment by vbuterin** [comment]
> Ethereum's inflation rate is not excessively high for a cryptocurrency that is only a few years old. Bitcoin's was much higher at the equivalent stage in its life.

Yes, but Ethereum is a _mature_ cryptocurrency. We have professionalized mining, a large ecosystem, a $30b market cap, etc etc. So @kybarnet 's point that bitcoin is only giving 4% issuance is actually quite reasonable.

> Ethereum's inflation structure is already disinflationary.

Yes, but very slowly so. Under the current scheme, assuming bomb delay but no reduction, inflation reduces to 4% only after 15 years.

> Useful currencies tend to be inflationary because inflation encourages people to use rather than hoard currency. Does the Ethereum Foundation want Ether to be used or hoarded?

Ether was never intended as a medium of exchange in consumer transactions. People may use it as such (or for that matter as store of value) but it's not the purpose that ETH is designed for. Also, even if it was, I would argue that "inflation encourages hoarding rather than spending" is a misreading of economic theory. The preference toward mild inflation in fiat currencies exists because of things like downward wage stickiness, basically economic frictions where _nominal_ downward price adjustments are harder to make than upward ones, and so it's optimal for the unit of account to always slightly move downwards to adjust for this friction. Cryptocurrencies are super-volatile, so this doesn't really apply.

Responding to "people will want to hoard" more directly, it's important to note that people never hoard literally 100% of their wealth in the asset that rises fastest; they want to have other assets for liquidity, because they want to buy things, for risk management etc etc, so it's not at all clear that if the medium of exchange rises in value more quickly spending will go down. There are two ways to hold more ETH: (i) buy more, (ii) spend less. If it's worth $0.1 for someone to send a transaction, then they will pay the $0.1 tx fee, and if they want to hoard more ETH again, they'll just buy the extra $0.1 of ETH. The usual argument brought up by goldbugs is "electronics get 2x better every 3 years, but people still buy laptops and phones"; I think that applies fully here.

> Miners and speculators often have different incentives, but miners are the ones who, in my opinion, contribute far more to the network. I think that the policies of the Foundation should consider their interests over those of speculators when there are disputes.

Actually, we are treating miners' and speculators' interests roughly equally. Giving the miners an extra $1b/year of revenue doesn't give them $1b/year of profit; maybe something like $100m. OTOH saving $1b of funds for the ecosystem actually does save $1b of resources. So unless we need to spend that $1b to ensure security, there's no reason to spend it.

Also regarding speculators vs miners, I'll echo Vlad's tweet here: https://twitter.com/VladZamfir/status/890230006902915072

> The purpose of block rewards is not just security. Block rewards help with currency distribution, liquidity, and price discovery. Only about 23% of Ethereum was mined, with the rest having been distributed in the ICO. Keeping the block reward at 5 for the near future would help ensure a fair distribution of Ether for new investors who are interested in the technology but who don't want to buy from speculators who wish to drive up the price.

This is 2017. If this was still the era of highly democratized CPU mining, then I would be quite a bit more sympathetic to this argument. Unfortunately, most mining is now professionalized, even with GPUs, so it's not clear that mining increases decentralization of ETH holdings by that much.

**Comment by alexvandesande** [comment]
@TheCryptoMines  pity you could not be contacted in time to attend representing the miner's perspective. You have some interesting points, so let me try to counter-argue:

* **The bomb shouldn't be taken in consideration** Maybe you're right that the bomb should have been delayed long before it became active, but it hasn't and has been active for some time. So not reducing it now would effectively mean a 66% increase in issuance.  Also, some people might argue that the community expected the issuance of ether to be reduced by now by PoS that was defined in the creation of ethereum, and if PoS isn't live at least the reduction should be.

* **Inflation:** I do not believe it is the job of major client developers (Ethereum foundation is not the only one with developers contributing to the major clients and making those decisions) to debate the rate or merits of inflation or deflation. There's a lot of bad economics there too, because inflation in traditional economics is usually measured in purchasing power of goods, not in total amount of currency in circulation.  Purchasing power of cryptocurrency(which determines hoarding, vs usage) varies wildly and is not determined by rate of issuance

* **Miners vs Coin Holders** Coin holders are not very affected by it. The total amount of ether in circulation with 3 versus 5 ether per block will differ by less than 5%

* **Distribution** Ether is highly liquid, there's tons of it on exchanges and markets. I would be curious to know how much of the original presale contributions remain with those participants. I would say probably not that much, so it's not like they can lobby and control the price. I would say that projects that pay in ether for other kinds of service, like golem, swarm, etc, have a much higher potential in impacting the access of ether to the common user

**Comment by SCBuergel** [comment]
Many great points here but let me add a very much down-to-earth comment regarding issuance / ETH distribution / price increase:

@TheCryptoMines 
> Keeping the block reward at 5 for the near future would help ensure a fair distribution of Ether for new investors who are interested in the technology but who don't want to buy from speculators who wish to drive up the price.

Precisely because of the Ether price shooting up massively we have seen a huge increase in public interest. The number of people that contact me personally regarding Etherium [sic] simply due to "hey I see you can make 10x in 6 months" seems to be significantly outnumbering the people that see the true revolutionary potential in this technology. This is in line with the increase of exchange users (and massive backlogs there). Those folks do not have a Github account or even know about this debate. If we like it or not, the interest in Ether is drastically increasing because the price increased drastically. Now you have those guys investing, some of them significantly - remember that there are so many more meat world millionaires than crypto millionaires, many are coming in now.

What I'm getting at: 
1) Price increases currently increase distribution of Ether among many people (instead of decreasing it "because Ether is so pricy now").
2) These guys are mostly speculators now but might be real users tomorrow (this is how I got into BTC a few years back and quit my job to work full time on blockchain later).
3) By directly or indirectly driving up the price, we are increasing bubble potential but also potential user numbers, how far are we willing to take this? There is probably a sweat spot somewhere - maybe at 3 ETH/block
4) Any discussion here has a massive financial impact that has to be taken seriously and not treated like a "technicality", you do a good job so far, keep it up!

**Comment by kybarnet** [comment]
@vbuterin Just to clarify I few points that will help you in making your presentations.

Regarding your post within this thread, it is very good. However, I recall somewhere some statements being made akin to X, which I will try to assist in clarifying the language for talking purpose.

'savings' and 'reductions' - Generally speaking, people will associate the words savings and reductions with Dollar value savings. It is fairly important when making such statements to name a price, such as "moving from 5 ETH to 3 ETH will create a savings, so long as the price remains below $300", etc. A good example is Bitcoin, which moved from 8% to 4% rates of inflation. The 'savings' and 'reduction' actually amounted to paying miners triple what they were getting paid before. A better phrase in my opinion is a 50% increase in efficiency, or financial efficiency, if you want to be precise. There will literally never be a dollar value 'savings' or 'reduction' in dollar values spent, only a improvement in financial efficiency, or a decrease in percentages waste, whichever is preferred. But maybe that's splitting hairs... 

'the future price is unknowable', etc. - There are only 3 ways that money leaves the Eth system:

* Founders cashing out, paying for expenses, Devs cashing out (genesis coins)

* Equipment Rentals, Miner Bonuses

* Profit taking, wealth reallocation, etc.

We can quantify the Founders cashing out, but generally let's say this is $100 Million a year or less. Secondly, we can quantify 'profit takers', which for simplicity we can say only occurs after an increase in value (at a stable price, there are no profit takers, reallocation). Lastly, we can quantify Miner Bonuses as roughly $2 Billion at $300.

So we got money leaving the system of Profit Takers + $2.1 Billion ($2 Billion miner bonuses, $100 Million Foundation spend). This is where the math gets tricky, but simple to understand at the same time.

If the price is stable, say $300, and there are no profit takers, what is the DEMAND for Ethereum? It is equal exactly to costs, or $2.1 Billion annually. What happens if costs are reduced by $1 Billion, which forces a $1 Billion buy order on the market that must become fulfilled by profit takers? The price must go up, until profit takers are forced to reallocate to appropriate values, or the expenses increase to meet demand.

In the practical, real world, demand for greater level of financially efficient assets far outpaces the improvements in efficiency, and the total spent on Equipment Rentals for Ethereum would actually increase, significantly, probably 3 times more than the current $ spend, should the financial efficiency reach 4 to 2%.

While I agree with you that it is a waste of resources, in truth, should the rate of inflation fall from 8% to say 2%, this would likely lead to Ethereum paying miners $2 Billion -> $10 Billion or similar, within a few months. I could counter global warming concerns with global financial stability, perhaps, but the reality is that as the crypto market expands, so must the miner spend, until PoS technical achievements or through other artificial limitations (master noding, limiting the number of miners, etc).

So while many miners will argue with the presumption that an increase in financial efficiency will lead to a reduction in Miner Bonuses, it is factually incorrect. And likewise, many will argue that increased financial efficiency will lead to a dollar value savings. This again is also incorrect.

The only real question becomes how long does it take the market to respond, and how many profit takers must evacuate in-between such numbers. In a planned economy, like Bitcoin, the market responded ahead of the adjustment, and Miners received larger bonuses immediately. As you pointed out on the call, if the adjustment was overly dramatic, like let's say from 5 ETH to 0.2 ETH, this would require a ton of profit taking to stabilize, and thus miners would actually receive less for a period of time. But even with 50% reductions, the markets will stabilize within about a month, even with the largest crypto. In planned economies, the markets stabilize in anticipation of changes, and thus miners actually see increased earnings after increased financial efficiency adjustments.

Thus in the end, this becomes the real issue. How long did the market have to respond to the proposed adjustment? If more than a month to 3 months, it will likely respond prior to the adjustment, and thus there will likely be an increase in expenditure immediately following an increase in financial efficiency.

Thus, it is my prediction that with 5 -> 3 ETH, and with the other valued improvements, that Metropolis will hit a price of $500 (easily, IMO), prior to the execution of the objective.

If there is still time, one could spend $100,000 or whatever it takes to start the 550 hours (23 days) of computer cycling in order to have on hand machines at 1.25 ETH, 2 ETH, 2.5 ETH, etc... Thus, at the time of the execution, one can pick between the set which is the most appropriate value based on market adjustments.

As mentioned previously, as Miner Bonuses are a highly significant portion of the only ways in which money can leave a stable system (approximately 95% IMO), a system paying out $1 Billion annually is far more stable than a system paying $5 Billion annually, as it requires less Demand of New Investment.

Or put another way, effectively there are limitations upon the speed at which new money can flow into a system. With Ethereum, new money is currently flowing in about $2 Billion per year, or maybe $3 Billion if you account for the profit takers, price escalations. Should 3 ETH amount to $5 Billion at the time of execution, I would predict that the price will be unlikely to rise substantially, and thus one should discourage new investors from purchasing Ethereum at any value over $74 Billion or $700, between the time of Byzantium and the following update in one year (say).

Generally speaking however, the price of Ethereum is limited to the Dollar waste of Bitcoin. Bitcoin is currently wasting $3 Billion, which would put Ethereum at a price around $450, at market stability, with 3 ETH. Until Ethereum is below 4%, there is no reason it should be allowed to waste more money than Bitcoin, and the market would likely prefer the reverse (waste more money utilizing the asset with the greater rate of financial efficiency, etc).

Hope that helps you and Vlad in future analysis.

---

The comments below may be helpful in making stronger arguments in support of the positions you took within this thread.

> Also, even if it was, I would argue that "inflation encourages hoarding rather than spending" is a misreading of economic theory.

Your point is good, but another relevant point is that Crypto can only motivate the economy via increased Equipment Rentals of Computers, which is not an ideal way to spend 5% of the Economy. As a whole, maybe 0.01% is a suitable global value spend on Equipment Rental of Computers. If Crypto could price deflate, through issuance of housing, food, and medicine, then 5% would be suitable, as it would create more humans, which in turn create more value. - So IMO, the best counter argument is that the inflation mechanics within crypto are not perfectly inline with the propagation of life and duty, and thus increasing value is unavoidable without wasting, or destroying value simply to prevent the accumulation of value, but which serves no purpose.

If the objective it to maintain a $100 Price of Ethereum, the best way to do so is to sell Ethereum until the price falls to $100, and then burn all the money you acquired, so that it can not be reinvested into Ethereum. Effectively one does this through inefficiency, in such a way terrorism and negligence become one and the same, regarding outcomes.

> People never hoard literally 100% of their wealth in the asset that rises fastest; they want to have other assets for liquidity, because they want to buy things, for risk management etc etc.

This is a very astute point for understanding market mechanics. Regarding the point you are making, you are correct as well, but recall this metric as it applies to markets (as mentioned above regarding profit takers).

Another way to think about it is -

If you have 1% Ownership of USD, and want to perpetuate your wealth for Eternity, then your objective would be to acquire 1% Ownership in all future assets, regardless of current valuations. In such a way, even if Bitcoin took over the world, it wouldn't matter as you have maintained consistent level of ownership.

This is mainly to serve the point that the wealthy don't give a flip about stopping Bitcoin, or Crypto. It is much cheaper to simply buy an appropriate share of ownership using the wealth they currently have, in which case they profit equally from the increase of USD, or the increase in BTC. There is not *really* a secret underground ring of rich people ending crypto, they already bought their share of interest, using Dollars, and thus their status is protected from all possible outcomes, as you mentioned, diversified risk, etc.

Thus, one might end up cutting off their nose to spite their face, or however the saying goes. Which is to say destroying oneself to prevent the profiteering of others from your labor or ingenuity is only questionably effective. It essentially becomes reliant upon all others destroying themselves as well, which is improbable, fortunately.

Many often misrepresent the American Revolution as a rebellion to the crown, when it was at first only a suggestion.

Good luck and take care.

**Comment by kybarnet** [comment]
> The total amount of ether in circulation with 3 versus 5 ether per block will differ by less than 5%.

This makes me happy, thank you Alex.

**Comment by Souptacular** [comment]
Closing: https://github.com/ethereum/pm/blob/master/All%20Core%20Devs%20Meetings/Meeting%2023.md
